blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/EnumValidator.java

```
@@ -0,0 +1,167 @@
+            Future<ColumnList<Long>> enumHashValuesFuture = taskExecutor.submit(new Callable() {
+                @Override
+                public ColumnList<Long> call() throws Exception {
+                    return AstyanaxReader.getInstance().getColumnsFromEnumCF(locator);
+                }
+
+            });
+
+            // start async read for enum values of locator from elastic search
+            Future<List<SearchResult>> esEnumValuesFuture = taskExecutor.submit(new Callable() {
+                @Override
+                public List<SearchResult> call() throws Exception {
+                    return discoveryIO.search(locator.getTenantId(), locator.getMetricName());
+                }
+            });
+
+
+            Map<Long, String> enumStringValues = null;
+            List<SearchResult> esEnumValues = null;
+            try {
+                // values from cassandra db
+                ColumnList<Long> enumHashValues = enumHashValuesFuture.get();
+                if ((enumHashValues != null) && (enumHashValues.size() > 0)) {
+                    enumStringValues = AstyanaxReader.getInstance().getEnumValueFromHash(enumHashValues);
+                    log.debug(String.format("Enum values for metric %s: %s", locator.toString(), enumStringValues.toString()));
+                }
+                else {
+                    log.debug(String.format("No enum values found for metric %s", locator.toString()));
+                }
+
+                // values from elasticsearch
+                esEnumValues = esEnumValuesFuture.get();
+
+                validateThresholdAndIndex(locator, enumStringValues, esEnumValues);
+
+            } catch (InterruptedException e) {
+                log.error("Interrupted Exception during query of Enum metrics in EnumValidator", e);
+            } catch (ExecutionException e) {
+                log.error("Execution Exception during query of Enum metrics in EnumValidator", e);
+            }
```

Do we want to catch all exceptions here?